roles/splunk/tasks/configure_firewall.yml

```diff
@@ -0,0 +1,35 @@
+---
+- name: Ensure {{ firewall_service }}  package is installed
+  ansible.builtin.package:
+    name: "{{ firewall_service }}"
+    state: present
+  become: true
+
+- name: Ensure {{ firewall_service }} is Started and Enabled
+  ansible.builtin.systemd:
+    name: "{{ firewall_service }}"
+    state: started
+    enabled: true
+  become: true
+
+- name: Configure firewalld for Splunk
+  ansible.builtin.template:
+    src: firewalld_service.xml.j2
+    dest: /etc/firewalld/services/splunk.xml
+    backup: true
+    mode: 0644
+    owner: root
+    group: root
+  become: true
+  notify: reload firewalld
+  when: firewall_service == "firewalld"
+
+- name: Configure UFW for Splunk
+  community.general.ufw:
```

Comment:
What about adding a comment to UFW rules as well? Would also be more consistent since you are already using `<description>` in firewalld_service.xml.j2.

`comment: "{{ item.key }}"` should do the trick as the port var names is already fairly descriptive.

Reply:
Good suggestion - would love to have a XML comment in the Firewalld service that has the same concept - will this work if we're not iterating over key-value pairs though? Is `item.key` a reserved name when iterating over a list of variables? @jewnix any thoughts?

Reply:
@arcsector you are right we cant use `item.key` in a dictionary so we would have to define a `desc` key in the dictionary.
Ex `splunkweb_port: { protocol: "tcp", number: 8000, desc: "Splunk Web" }` 

We can implement `comment: ` as optional like this as UFW accepts an empty string as comment.

```yaml
- name: Add splunk port to UFW
      community.general.ufw:
        port: "{{ item.number }}"
        proto: "{{ item.protocol }}"
        comment: "{{ item.desc | default('') }}"
        rule: allow
        state: reloaded
      become: true
      loop: "{{ splunk_firewall_ports }}"

# TASK [Add splunk port to UFW] **************************************************************************************
# changed: [localhost] => (item={'protocol': 'tcp', 'number': 8000, 'desc': 'Splunk Web'})
# changed: [localhost] => (item={'protocol': 'tcp', 'number': 8089})
#
#
# $ sudo ufw status verbose
#
# To                         Action      From
# --                         ------      ----
# 8000/tcp                   ALLOW IN    Anywhere                   # Splunk Web
# 8089/tcp                   ALLOW IN    Anywhere   
```

Reply:
This would definitely be good if you had a non-standard value for ports, but I don't know if we want to complicate the dictionary more. So i do like the idea of making the default value for the description, so that sounds like a plan. I'll push that too.

Reply:
I'm all for adding descriptions.

Reply:
Done in https://github.com/splunk/ansible-role-for-splunk/pull/157/commits/4c7b5f3381933bc6473b3c48df1ee23cda16ef8e